core/trino-main/src/main/java/io/trino/memory/ClusterMemoryManager.java

```diff
@@ -174,18 +174,25 @@ public synchronized void addChangeListener(Consumer<MemoryPoolInfo> listener)
         changeListeners.add(listener);
     }
 
-    public synchronized void process(Iterable<QueryExecution> runningQueries, Supplier<List<BasicQueryInfo>> allQueryInfoSupplier)
+    public synchronized void process(Supplier<Stream<QueryExecution>> allQueries, Function<QueryId, Optional<QueryExecution>> executionInfoSupplier)
```

Comment:
Given that stream itself is lazy, why is there a need to use a Supplier to create the Stream ?

Reply:
You can't consume stream twice (it's like an iterator). In this case we need to read it twice